water_mark/water_mark.py

```diff
@@ -0,0 +1,303 @@
+            if 'Mask' in f:
+                python = 'python'
+                windowsMode = detect_windows_OS()
+                if not windowsMode:
+                    python = 'python3'
+                input_file = f
+                output_file = f"Binary{f}"
+                processDataset(
+                    src_file=input_file,
+                    dst_file=output_file,
+                    in_classes='<=45, <=100',
+                    out_classes='0, 1',
+                    default=0,
+                    nodata=True,
+                    output_format='GTiff',
+                    compression='COMPRESS=LZW'
+                )
+                os.remove(f)
+                old_name = os.path.join(out_dir, f"Binary_{f}")
+                new_name = os.path.join(out_dir, f)
+                os.rename(old_name, new_name)
+
+
+def copy_vv_vh_to_inputs(out_dir, data_dict):
+    for sar, vvvhband in data_dict.items():
+        shutil.copy(
+            os.path.join('inputs', vvvhband[0]),
+            os.path.join(out_dir, sar)
+        )
+        shutil.copy(
+            os.path.join('inputs', vvvhband[1]),
+            os.path.join(out_dir, sar)
+        )
+
+
+def make_masks(out_dir, data_dict):
+    count = 0
+    for sar, vvvhband in data_dict.items():
+        renamePath = os.path.join(out_dir, sar)
+        print(f"{count} masks made")
+        idw_main(
+            os.path.join('inputs', vvvhband[0]),
+            os.path.join('inputs', vvvhband[1])
+        )
+        count += 1
+        # rename 'mask-0.tif' to 'Mask_<sar>.tif'
+        os.rename(
+            'mask-0.tif',
+            os.path.join(renamePath, f"Mask_{sar}.tif")
+        )
+
+
+def tile(out_dir, tif_name, sar, mxm_tile_size, isMask):
+    label = 'temp'
+    if isMask:
+        label = 'Mask'
+    else:
+        label = 'Image'
+    tif = os.path.join(out_dir, sar, tif_name)
+    tifData = gdal.Open(tif)
+    xStep, yStep = mxm_tile_size, mxm_tile_size
+    xSize, ySize = tifData.RasterXSize, tifData.RasterYSize
+    count = 0
+    for x in range(0, xSize, xStep):
+        for y in range(0, ySize, yStep):
+            # fileName = '<Image|Mask>_<sar>_<0-9+>.tif'
+            fileName = f"{label}_{tif_name[:-4]}_{count}.tif"
+            output_file = os.path.join(out_dir, sar, fileName)
+            input_file = tif
+            gdal.Translate(
+                output_file,
+                input_file,
+                srcWin=[x, y, xStep, yStep],
+                format="GTiff"
+            )
+            count += 1
+
+
+def tile_vv_vh_mask(out_dir, mxm_tile_size):
+    for sar in os.listdir(out_dir):
+        for tif_name in os.listdir(os.path.join(out_dir, sar)):
+            if tif_name.endswith('VV.tif'):
+                tile(out_dir, tif_name, sar, mxm_tile_size, False)
+            elif tif_name.endswith('VH.tif'):
+                tile(out_dir, tif_name, sar, mxm_tile_size, False)
+            elif tif_name.startswith('Mask'):
```

Comment:
Since you're removing 5 characters, how about you check that all of those 5 characters are actually there
```suggestion
            elif tif_name.startswith('Mask_'):
```